File: Code/Disney/disney.xBandController/src/windows/Test/Disney.xBand.xBMS.Simulator/Dto/BusinessEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using System.Xml.Serialization;

namespace Disney.xBand.xBMS.Simulator.Dto
{
    [XmlRootAttribute(ElementName = "businessEvent", IsNullable = false, Namespace="")]
    public class BusinessEvent
    {
        [XmlIgnore]
        public string Payload { get; set; }

        [XmlElement(ElementName = "location", Order = 1)]
        public String Location { get; set; }

        [XmlElement(ElementName = "eventType", Order = 2)]
        public String EventType { get; set; }

        [XmlElement(ElementName = "subType", Order = 3)]
        public String SubType { get; set; }

        [XmlElement(ElementName = "referenceId", Order = 4)]
        public String ReferenceId { get; set; }

        [XmlElement(ElementName = "guestIdentifier", Order = 5)]
        public String GuestIdentifier { get; set; }

        [XmlElement(ElementName = "timeStamp", Order = 6)]
        public String TimeStamp { get; set; }

        [XmlElement(ElementName = "payload", Order = 7)]
        public System.Xml.XmlCDataSection PayloadCData
        {
            get
            {
                return new System.Xml.XmlDocument().CreateCDataSection(Payload);
            }
            set
            {
                Payload = value.Value;
            }
        }
        [XmlElement(ElementName = "correlationId", Order = 8)]
        public String CorrelationId { get; set; }
    }
}

File: Code/Disney/disney.xBandController/src/linux/JmsLib/src/main/java/com/disney/xband/jms/lib/entity/gff/BusinessEvent.java
package com.disney.xband.jms.lib.entity.gff;

import java.io.Serializable;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

/*
 <businessEvent>
    <location>GXP.BLUELANE</location>
    <eventType>Entitlement</eventType>
    <subType>BlueLane</subType>
    <referenceId>0776810827978534</referenceId>
    <guestIdentifier>0776810827978534</guestIdentifier>
    <timeStamp>2011-11-18T15:19:11Z</timeStamp>
    <payLoad><![CDATA[
        <blueLane>
            <xbandId>0776810827978534</xbandId>
            <gxpEntertainmentId>80010110</gxpEntertainmentId>
            <reason>{No Valid Park Admission || No Xpass || Early || Late || Different Entertainment}</reason>
            <tapTime>Fri Nov 18 15:19:11 EST 2011</tapTime>
            <facilityId>80007944</facilityId>
        </blueLane>
    ]]></payLoad>
    <correlationId>074321ff-5af5-44a5-85dd-fb8bf2b8980c</correlationId>
</businessEvent>


 */

@XmlRootElement(name="businessEvent")
public class BusinessEvent implements Serializable 
{

	private static final long serialVersionUID = 1L;
	
	private String location;

	private String eventType;
	
	private String subType;
	
	private String referenceId;
	
	private String guestIdentifier;


    private String facilityId;
	
	private String timeStamp;
	
	private String correlationId;
	
	private String payload;
    private String facilityOneSourceId;

    @XmlElement(name="location")
	public String getLocation()
	{
		return this.location;
	}
	
	public void setLocation(String location)
	{
		this.location = location;
	}
	
	@XmlElement(name="event_type")
	public String getEventType()
	{
		return this.eventType;
	}
	
	public void setEventType(String eventType)
	{
		this.eventType = eventType;
	}
	
	@XmlElement(name="subtype")
	public String getSubType()
	{
		return this.subType;
	}
	
	public void setSubType(String subType)
	{
		this.subType = subType;
	}
	
	@XmlElement(name="referenceid")
	public String getReferenceId()
	{
		return this.referenceId;
	}
	
	public void setReferenceId(String referenceId)
	{
		this.referenceId = referenceId;
	}
	
	@XmlElement(name="guest_identifier")
	public String getGuestIdentifier()
	{
		return this.guestIdentifier;
	}
	
	public void setGuestIdentifier(String guestIdentifier)
	{
		this.guestIdentifier = guestIdentifier;
	}
	
	@XmlElement(name="timestamp")
	public String getTimeStamp()
	{
		return this.timeStamp;
	}
	
	public void setTimeStamp(String timeStamp)
	{
		this.timeStamp = timeStamp;
	}
	
	@XmlElement(name="correlationId")
	public String getCorrelationId()
	{
		return this.correlationId;
	}
	
	public void setCorrelationId(String correlationId)
	{
		this.correlationId = correlationId;
	}
	
	@XmlElement(name="payLoad")
	public String getPayLoad()
	{
		return this.payload;
	}
	
	public void setPayLoad(String payload)
	{
		this.payload = payload;
	}

    @XmlElement(name = "facilityid")
    public String getFacilityId() {
        if(facilityId == null) {
            return facilityOneSourceId;
        }
        return facilityId;
    }

    public void setFacilityId(String facilityId) {
        this.facilityId = facilityId;
    }

    @XmlElement(name = "facilityonesourceid")
    public String getFacilityOneSourceId() {
        return facilityOneSourceId;
    }


    public void setFacilityOneSourceId(String value) {
        this.facilityOneSourceId = value;
    }


}
